### PR TITLE
Fix issue 18046

### PR DESCRIPTION
**Pull request by luca-i**
This patch should fix issue [18046](https://github.com/ziglang/zig/issues/18046).

I wasn't able to get a failing test with `std.testing.expectFmt`, as suggested, because the overflow error shows up only during comptime evaluation, so I added `std.testing.expectComptimeFmt`.

Not sure this is the right way to go... this is my very first attempt to submit a pull request.

Thank you in advance,
Luca-

### CODE REVIEW

File: lib/std/fmt.zig [hunk 1]
```diff
@@ -2059,6 +2059,8 @@ test "comptimePrint" {
     @setEvalBranchQuota(2000);
     try std.testing.expectEqual(*const [3:0]u8, @TypeOf(comptimePrint("{}", .{100})));
     try std.testing.expectEqualSlices(u8, "100", comptimePrint("{}", .{100}));
+    try std.testing.expectComptimeFmt("30", "{d}", .{30.0});
+    try std.testing.expectComptimeFmt("30.0", "{d:3.1}", .{30.0});
```
Comment (squeek502) [inline lib/std/fmt.zig:2063]: I believe this should be equivalent without the need for the new `expectComptimeFmt` function:
```suggestion
    try std.testing.expectEqualStrings("30", comptimePrint("{d}", .{30.0}));
    try std.testing.expectEqualStrings("30.0", comptimePrint("{d:3.1}", .{30.0}));
```

Comment (luca-i) [reply lib/std/fmt.zig:2063]: Thank you @squeek502 for pointing this out. I'll update my pull request.

Comment (luca-i) [reply lib/std/fmt.zig]: (@squeek502  uhm.. I just realized I could simply merge your commit, right? Sorry, this is my very first pull request)

Comment (squeek502) [reply lib/std/fmt.zig]: No worries, what you did is good.

**Pull request merged**